backend/src/test/java/ca/gov/dtsstn/cdcp/api/service/UserServiceTests.java

```diff
@@ -140,4 +147,42 @@ void testGetUserById() {
 		assertThat(userService.getUserById("00000000-0000-0000-0000-000000000000")).isNotEmpty();
 	}
 
-}
+	@Test
+	@DisplayName("Test userService.verifyEmail(..) with valid input")
+	void testVerifyConfirmationCode_Valid() {
+		final var creationDate = LocalDateTime.now().minusDays(73).toInstant(ZoneOffset.UTC);
+		final var expiryDate = LocalDateTime.now().plusDays(288).toInstant(ZoneOffset.UTC);
+		final var codeValue = "code value";
+		final var confirmationCode = new ConfirmationCodeEntityBuilder().code(codeValue).createdDate(creationDate).expiryDate(expiryDate).build();
+		final var mockUser = new UserEntityBuilder().confirmationCodes(Collections.singleton(confirmationCode)).build();
+
+		when(userRepository.findById(any())).thenReturn(Optional.of(mockUser));
+
+		assertThat(userService.verifyEmail(codeValue, userId)).isTrue();
+	}
+
+	@Test
+	@DisplayName("Test userService.verifyEmail(..) with null code")
+	void testVerifyConfirmationCode_NullCodeValue() {
+		assertThrowsExactly(IllegalArgumentException.class, () -> userService.verifyEmail(null, userId), "code is required; it must not be null");
+	}
+
+	@Test
+	@DisplayName("Test userService.verifyEmail(..) with empty code string")
+	void testVerifyConfirmationCode_EmptyCodeValue() {
+		assertThrowsExactly(IllegalArgumentException.class, () -> userService.verifyEmail("", userId), "code is required; it must not be null");
+	}
+
+	@Test
+	@DisplayName("Test userService.verifyEmail(..) with mismatched input")
+	void testVerifyConfirmationCode_Mismached() {
```

Comment:
For consistency, these unit test methods should be renamed to `testVerifyEmail...()` since the method being tested has been renamed.

Reply:
Makes sense. Implemented in commit 70d88ff21299e8b6f77785dd0b22d60b10f1a733 .